bugfix: avoid endless loop

robertnurnberg opened the pull request:
Fixes a typo and a separate bug that can lead to an endless loop: if the `move()` function sets `skip` to true for a move with a comment. As happened at https://github.com/vondele/fastpopular/issues/26.

Disservin commented:
can you also add the pgn which caused this as a test? since now the parsing will actually complete

robertnurnberg commented:
Yes, I can. So master would get into an endless loop for this test, right? Or is there some time cut-off to allow it to fail?

robertnurnberg commented:
I added the test pgn. I turns out my original fix did not cover all the cases, so I also changed `src/pgn.cpp`.

Beware that master will hang on the test.

Also added a line to the readme.
